### PR TITLE
PORTF-989 Added missing Siklu code in i2c_imx_isr

### PR DESCRIPTION
**Pull request by yba-siklu**
## Description
The i2c-imx.c i2c_imx_isr was missing a callback to
i2c_imx_isr_slave_event. Copied the missng code from the 4.9.11 kernel.
This enables the I2C data transfer to userspace whose absence prevented
us from seeing that the RF link is actually up.

## Tests
Tested using #define DEBUG in i2c-imx.c, tested that RF link is up.

## References
JIRA: PORTF-989

### CODE REVIEW

**Comment by dima1308** [comment]
@yba-siklu can we delete debug lines/commented out code from the PR? Should we merge this PR before sending the version to QA?

**Comment by yba-siklu** [comment]
> @yba-siklu can we delete debug lines/commented out code from the PR? Should we merge this PR before sending the version to QA?

I would recommend against deleting the debugp lines because they provide an indication about how to debug the code. Without them the next engineer has to learn the code again and will see debug output that is different from the previous engineer. I regard the dev_dbg and debugp lines as an integral part of the code.

Regarding comments in the code. I generally do not delete other engineers' comments unless there is a specific decision to do so. If there are lines of code that someone else added and commented out, I also generally do not touch them unless it is part of a code cleanup task and I have the time to really think about why the lines might have been left in the code but commented out.

At this point in the project I think that it is important to get the code tested and we can clean up the comments later. I suspect that Baruch's new PHY code does not need any kernel modifications, so in any event this code has limited lifetime.

**Comment by dima1308** [comment]
> I would recommend against deleting the debugp lines because they provide an indication about how to debug the code. Without them the next engineer has to learn the code again and will see debug output that is different from the previous engineer. I regard the dev_dbg and debugp lines as an integral part of the code.
> 
> Regarding comments in the code. I generally do not delete other engineers' comments unless there is a specific decision to do so. If there are lines of code that someone else added and commented out, I also generally do not touch them unless it is part of a code cleanup task and I have the time to really think about why the lines might have been left in the code but commented out.
> 
> At this point in the project I think that it is important to get the code tested and we can clean up the comments later. I suspect that Baruch's new PHY code does not need any kernel modifications, so in any event this code has limited lifetime.

OK, understood, let's wait for @baruchsiach review.
Should we merge this PR before sending the version to QA?

**Comment by yba-siklu** [comment]
> > I would recommend against deleting the debugp lines because they provide an indication about how to debug the code. Without them the next engineer has to learn the code again and will see debug output that is different from the previous engineer. I regard the dev_dbg and debugp lines as an integral part of the code.
> > Regarding comments in the code. I generally do not delete other engineers' comments unless there is a specific decision to do so. If there are lines of code that someone else added and commented out, I also generally do not touch them unless it is part of a code cleanup task and I have the time to really think about why the lines might have been left in the code but commented out.
> > At this point in the project I think that it is important to get the code tested and we can clean up the comments later. I suspect that Baruch's new PHY code does not need any kernel modifications, so in any event this code has limited lifetime.
> 
> OK, understood, let's wait for @baruchsiach review. Should we merge this PR before sending the version to QA?

1. We will merge this PR to repo linux-nxp-branches as soon as Baruch approves but not change the commit hash for Linux in the PORTF-827-Unified-BSP-H-merge-master branch yet.
2. The portfolio PR https://github.com/siklu/portfolio/pull/250 is a topic branch PR that corresponds to this linux-nxp-branches PR and can be used to build an image for testing, or Avi can use the same uimage file that I put in T/newbsp/2022-06-09/siklu-uimage-nxp-10.5.0-18324-65bc72127c for Shmulik just now
3. After Shmulik tests, Avi will test
4. After Avi tests, I will update the topic PR https://github.com/siklu/portfolio/pull/250 to be a regular PR with updated commit hash to merge to the PORTF-827-Unified-BSP-H-merge-master branch
5. At some future time we will either merge PORTF-827-Unified-BSP-H-merge-master to master or not.

File: drivers/i2c/busses/i2c-imx.c
Comment (baruchsiach) [inline drivers/i2c/busses/i2c-imx.c:20]: Since there is no way to comment on the commit log, I'll comment here.

As far as I know, the RF link up indication is accessible from the modem control API that communicates with the K60 MCU over UART. The I2C interface does not provide link up indication. I2C only provides information about the link quality, modulation profile, and transfer rates. This allows the host software running on the i.MX6 SoC to decide on the best modulation.

**Review by baruchsiach — APPROVED** [review]
Other than slight change in commit log, looks good to me.

File: drivers/i2c/busses/i2c-imx.c
Comment (yba-siklu) [reply drivers/i2c/busses/i2c-imx.c:20]: host/app/modem/modem_nxp/modem/statistics.cpp contains the current state machine for determining link status depending on I2C events that the i.MX6 reads from the k60.

The "new BSP" effort was defined as a porting effort only, not a rewrite of the PHY/modem code. The rewrite of the PHY/modem code task was assigned to you AFAIK.

**Pull request merged**
